x/concentrated-liquidity/README.md

```diff
@@ -234,12 +234,18 @@ One draw back of this implementation is the requirement to create many ticks tha
 
 - **Request**
 
-This message allows LPs to provide liquidity between `LowerTick` and `UpperTick` in a given `PoolId.
+This message allows LPs to provide liquidity between `LowerTick` and `UpperTick` in a given `PoolId`.
 The user provides the amount of each token desired. Since LPs are only allowed to provide
 liquidity proportional to the existing reserves, the actual amount of tokens used might differ from requested.
 As a result, LPs may also provide the minimum amount of each token to be used so that the system fails
 to create position if the desired amounts cannot be satisfied.
 
+Three KV stores are initialized when a position is created:
+
+1. `Position ID -> Position` - This is a mapping from a unique position ID to a position object. The position ID is a monotonically increasing integer that is incremented every time a new position is created.
+2. `Owner | Pool ID | Position ID -> Position ID` - This is a mapping from a composite key of the owner address, pool ID, and position ID to the position ID. This is used to keep track of all positions owned by a given owner in a given pool.
```

Comment:
Is that the case?

Why do we have `Owner | Pool ID | Position ID -> Position ID` and not `Owner | Pool ID -> list of Position ID`?

Or removing the position id from the value. Having duplicates seems unnecessary

Reply:
Maybe we can revisit this one in a follow-up issue

Reply:
If you do `Owner | Pool ID -> List of Position IDs` then you cant query all position IDs for an owner. The reason its `Owner | Pool ID | Position ID -> Position ID` is because the Position ID in the key is just used as a nonce to not write over data and have separate entries.

Reply:
> then you cant query all position IDs for an owner

Why not?

The process is:
1. Iterate the owner prefix to collect the lists of position IDs within all pools
2. merge position IDs as you iterate into a single list
3. Return the list

The problem with this right now is that we are duplicating a large amount of data in the store. Not only do we duplicate position ids in key and the value but also for each position ID, we have a combination of 3 fields "Owner | Pool ID | Position ID"

It's the difference between:

Right now:
```
addr1|1|1->1
addr1|1|2->2
addr1|2|3->3
addr2|1|4->4
addr2|1|5->5
```

Suggested:
```
addr1|1|->[1, 2]
addr1|2|->[3]
addr2|1|->[4,5]
```

Imagine this but at a much larger scale.

I still think there is no reason to block the PR on this optimization but I think we should keep track of this in an issue to further analyze

Reply:
Wait yeah, you are correct, thanks for breaking this down! Will create an issue for this, I agree the key format should be the way you explained